sw/device/silicon_creator/lib/boot_data.c

```diff
@@ -160,6 +160,63 @@ static rom_error_t boot_data_entry_read(uint32_t page_base, size_t index,
   return kErrorOk;
 }
 
+/**
+ * Populates the boot data entry at the given page and index.
+ *
+ * @param page_base Page base address in bytes.
+ * @param index Index of the entry to write in the given page.
+ * @param boot_data Entry to write.
+ * @return The result of the operation.
+ */
+static rom_error_t boot_data_entry_write(uint32_t page_base, size_t index,
+                                         const boot_data_t *boot_data) {
+  const uint32_t addr = boot_data_entry_address_get(page_base, index);
+  boot_data_buffer_t buf;
+  memcpy(&buf, boot_data, sizeof(boot_data_t));
+  // TODO(#8777): Update error handling after switching to silicon_creator
+  // driver.
+  if (flash_write(addr, kInfoPartition, buf.data, kBootDataNumWords) != 0) {
+    return kErrorBootDataFlash;
+  }
+  if (flash_read(addr, kInfoPartition, kBootDataNumWords, buf.data) != 0) {
+    return kErrorBootDataFlash;
+  }
+  if (memcmp(&buf, boot_data, sizeof(boot_data_t)) != 0) {
+    return kErrorBootDataFlash;
+  }
+  return kErrorOk;
+}
+
+/**
+ * Invalidates the boot data entry at the given page and index.
+ *
+ * This function sets the `identifier` field of the given entry to
+ * `kBootDataInvalidatedIdentifier` which will cause both the identifier and the
+ * digest checks to fail in subsequent reads.
+ *
+ * This function must be called only after the new entry is successfully
+ * written since writes can potentially be interrupted.
+ *
+ * @param page_base Page base address in bytes.
+ * @param index Index of the entry to invalidate in the given page.
+ * @return The result of the operation.
+ */
+static rom_error_t boot_data_entry_invalidate(uint32_t page_base,
+                                              size_t index) {
+  static_assert(offsetof(boot_data_t, identifier) % sizeof(uint32_t) == 0,
+                "`identifier` must be word aligned.");
+
+  const uint32_t addr = boot_data_entry_address_get(page_base, index) +
+                        offsetof(boot_data_t, identifier);
+  // TODO(#8777): Update error handling after switching to silicon_creator
+  // driver.
+  const uint32_t val = kBootDataInvalidatedIdentifier;
+  if (flash_write(addr, kInfoPartition, &val, 1) != 0) {
```

Comment:
This should be already done, please see this [line](https://github.com/lowRISC/opentitan/blob/f5d8bbc5ac76df9e0fcbb1aa234f3c2043b2a3bf/sw/device/silicon_creator/lib/boot_data.h#L66) for the definition and this [line](https://github.com/lowRISC/opentitan/blob/f5d8bbc5ac76df9e0fcbb1aa234f3c2043b2a3bf/sw/device/silicon_creator/lib/boot_data.h#L102) for the assertion.

Invalidated identifier value is currently `0` so that we can program any non-zero identifier value to zero (in case we need to change the identifier). I guess we can define it to be anything as long as it doesn't require a 0->1 transition.

Reply:
Resolving, please reopen if you have additional comments.